pkgs/by-name/li/little_boxes/package.nix

```diff
@@ -0,0 +1,41 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, installShellFiles
+, callPackage
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "little_boxes";
+  version = "1.10.0";
+
+  src = fetchFromGitHub {
+    owner = "giodamelio";
+    repo = "little_boxes";
+    rev = version;
+    hash = "sha256-Quh09K5meiA39ih/orJWF2WfkuZdymxub1dZvns/q3E=";
+  };
+
+  cargoVendorDir = "vendor";
+
+  # Install manpages/completions
+  nativeBuildInputs = [ installShellFiles ];
+  postInstall = ''
```

Comment:
Since you are the upstream developer, what about doing this post-installation in upstream itself?
Can Rust machinery install these things correctly, possibly by adding some special instruction?

Reply:
Ya, I tried to do that initially, but there is no easy way that I could find to provide a the manpages/completions while also leaving the source for Nix to build.

It seems like it will end up as a script either way, and I would prefer to have in the Nix side where there is good tooling vs a one off shell script to build a better source for the Nix package.

Reply:
Well, then there are more serious problems, since

```
extrasPath=$(ls -d target/*/release/build/little_boxes-*/out/)
```

is not reliable. It should have better ways to do this.

Reply:
I was able to improve this a bit, but it still needs a wildcard for one hash. I tried to inline the hash (we could just update it with the version each time), but it depends not just on the source but also the arch `rustc` is building on.

I think it should be pretty stable though, because the whole `target` directory is created fresh during the build, so there should only ever be one `little_boxes-*` directory for the shell to glob on.

The only other alternative I could find is depending on an [external tool](https://github.com/Techcable/cargo-outdir) that is a bit of a hack and not well maintained. I will be following rust-lang/cargo#7546 for a better official solution eventually. Hopefully it is something that can be added to `rustPlatform.buildRustPackage` so every postInstall just has an environment variable for it.

Reply:
Well, let's move on, then.